File: core/src/remove/tanks/game/physics/fixture/hitbox/HitBoxPrefab.java
package remove.tanks.game.physics.fixture.hitbox;

import com.google.common.base.MoreObjects;
import remove.tanks.game.physics.fixture.FixturePrefab;
import remove.tanks.game.utility.Prefab;

/**
 * @author Mateusz Długosz
 */
public final class HitBoxPrefab extends Prefab {
    private final String id;
    private final FixturePrefab fixturePrefab;

    public HitBoxPrefab(String id, FixturePrefab fixturePrefab) {
        this.id = id;
        this.fixturePrefab = fixturePrefab;
    }

    public String getId() {
        return id;
    }

    public FixturePrefab getFixturePrefab() {
        return fixturePrefab;
    }

    @Override
    public String toString() {
        return MoreObjects.toStringHelper(this)
                .add("id", id)
                .add("fixturePrefab", fixturePrefab)
                .toString();
    }
}

File: tools/docs-generator/lib/physics/fixture/hit_box_prefab.py
class HitBoxPrefab(object):
    def __init__(self, prefab_id, fixture_prefab):
        self.id = prefab_id
        self.fixture_prefab = fixture_prefab

    def get_id(self):
        return self.id

    def get_fixture_prefab(self):
        return self.fixture_prefab

    def __str__(self):
        return "HitBoxPrefab(id={}, fixture_prefab={})".format(self.id, self.fixture_prefab)
